src/Microsoft.ApiDesignGuidelines.Analyzers/CSharp/CSharpIdentifiersShouldNotContainUnderscores.cs

```diff
@@ -11,5 +13,12 @@ namespace Microsoft.ApiDesignGuidelines.Analyzers
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     public sealed class CSharpIdentifiersShouldNotContainUnderscoresAnalyzer : IdentifiersShouldNotContainUnderscoresAnalyzer
     {
+        internal override void GetSyntaxNodeDiagnostics(CompilationStartAnalysisContext compilationStartAnalysisContext)
```

Comment:
You can probably just have an abstract property `ImmutableArray<SyntaxKind> SyntaxKinds {get;}` in `IdentifiersShouldNotContainUnderscoresAnalyzer` and avoid this abstract method